### ISSUE TITLE
[Bug]: No notifications or unread messages icon showing for Telegram (Z version) 

### ISSUE THREAD

**Issue by PlaceboPRS** [ferdium/ferdium-app#1009]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/ferdium/ferdium-recipes/blob/develop/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-recipes/blob/develop/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/ferdium/ferdium-recipes/issues) for a feature request that matches the one I want to file, without success.

### Ferdium Version

6.0.0 Nightly 23

### Recipe Name

Telegram

### Recipe Version

3.3.0

### Steps to reproduce

1. Install Telegram recipe
2. login to telegram
3. have people message you 
4. wonder why there's no notifications/unread message icons

### Expected Behavior

I expect to receive notifications and/or see an unread message icon

### Actual Behavior

I don't receive notifications and/or see an unread message icon

### Screenshots

![image](https://user-images.githubusercontent.com/23120453/166256243-93b22f2e-e547-4901-8919-ab5cb44c485f.png)

**Comment by vraravam** [comment]
Unable to reproduce this - I use Telegram, and I can see unread counts

**Comment by mydarkstar** [comment]
I also have trouble seeing an unread badge on the Telegram Web 'Z' version.
They seem to work fine with the 'K' version, however. @PlaceboPRS, could it be that you're using the 'Z' version?

I also pointed out that letting Telegram randomly choose one of the two versions can be detrimental to the user experience a bit ago [here](https://github.com/ferdium/ferdium-recipes/issues/13#issuecomment-1120349869).

Besides, all of your chats seem to be muted. Therefore you shouldn't be shown an unread badge anyway, no?

**Comment by PlaceboPRS** [comment]
Yes I was using Z because K wastes so much space:

![image](https://user-images.githubusercontent.com/23120453/167286320-a5acebb5-4cf7-4ff8-a78d-da423997ecf2.png)

I don't let Telegram randomly choose versions I chose the one that used the space best ;)

And no of course all my chats are not muted, that would be very dumb to mute all chats then wonder why I'm not getting notifications! I have a bunch of group chats muted but all my individual chats aren't muted :)

**Comment by mydarkstar** [comment]
Alright! I looked into this issue, and it seems like the following lines are broken on the 'WebZ' version:
https://github.com/ferdium/ferdium-recipes/blob/18801ed0c02627e87639dc0848cab44dacc18be2/recipes/telegram/webview.js#L8-L10

The returned value from this query is `'Telegram Web'` on the 'WebZ' version, while the script expects the string to contain `WebZ`. On the 'WebK' version the query still works, as it returns `'Telegram WebK'`.

The query could be changed to check for `WebK` and assume it's `WebZ` otherwise.
Another way of detecting the version may be using `document.location.pathname`, which should be either `/z/` or `/k/`. This approach is not as elegant, though.

**Comment by PlaceboPRS** [comment]
Hi, sorry but this isn't fixed, I'm using the Telegram recipe 3.3.1 and with Z version I'm still not getting notifications/unread counts.

**Comment by vraravam** [comment]
reopening

**Comment by vraravam** [comment]
Though I reopened this ticket, I have to say that I use Telegram, and I get notifications and unread counts. im on macos and the K version of Telegram

**Comment by PlaceboPRS** [comment]
> Though I reopened this ticket, I have to say that I use Telegram, and I get notifications and unread counts. im on macos and the K version of Telegram

Yes K version gets notifications, Z version doesn't. I use Z because K badly wastes/uses space.

I'm using Windows 11.

**Comment by mydarkstar** [comment]
Heya, on recipe v3.3.3, Telegram Web Z has been showing notification badges properly.

For groups and channels, it adds a blue dot:
![screenshot1](https://user-images.githubusercontent.com/17343993/175999267-ea84d3c7-1c75-450e-ab50-1de10041ad82.jpg)

For DMs, it shows a number:
![screenshot2](https://user-images.githubusercontent.com/17343993/175999330-0867a23e-e691-456f-99ad-05c9081c63e9.jpg)

Muted chats are ignored. This issue does not seem to persist on either the K or Z versions when using Ferdium Beta 2!
Please let us know if this is still an issue to you, @PlaceboPRS. Otherwise, I'd say we can close this.

**Comment by jasmines** [comment]
Unread count stopped working today, with 6.0.0 nightly 80

**Comment by ilopmar** [comment]
Telegram notifications were working until a few days ago. I updated to nightly.80 and they are not displayed when using Telegram K. If I switch to Telegram Z they are displayed.
I've tried also with beta2 and the behaviour is the same :(

P.S: I'm running Linux Mint with Cinnamon desktop.

**Comment by jasmines** [comment]
how can I switch to Z?

**Comment by ilopmar** [comment]
You can do it from the "hamburger" menu: 

![image](https://user-images.githubusercontent.com/559192/176618349-a76aa2dc-4bb1-47ff-a93d-2bfcd16107aa.png)


The problem is that you can't set that version as default. At least I don't know how to do it.

**Comment by MadelineProtoFan** [comment]
Sound notifications do not work even in Z version

**Comment by MadelineProtoFan** [comment]
> Telegram notifications were working until a few days ago. I updated to nightly.80 and they are not displayed when using Telegram K. If I switch to Telegram Z they are displayed. I've tried also with beta2 and the behaviour is the same :(
> 
> P.S: I'm running Linux Mint with Cinnamon desktop.

The same. 2 days ago everything was working :(

**Comment by MadelineProtoFan** [comment]
Notifications are showing. But there is no sound for them :( Using the latest 101 version :(
Do not have sound both K and Z version (everywhere the sound in on)

**Comment by MadelineProtoFan** [comment]
I can send the money to the developer. Please contact me and i will send the money for fixing sound notifications in telegram web! I cannot work without them

**Comment by MadelineProtoFan** [comment]
Please fix sound notifications in telegram. A month past

**Comment by MadelineProtoFan** [comment]
Today after update K version started to send sound notifications!
Everything is working. It was not a Ferdium problem. Telegram team fixed the sound notification problem
WHAT A GOOD DAY :)

**Comment by SpecialAro** [comment]
Hello @PlaceboPRS, is this issue still active?

**Comment by carlosala** [comment]
At least in Z version, this is still happening. No notifications at all. K version works as expected

**Comment by carlosala** [comment]
And counter in Z version is not correct, it's showing random numbers./

**Comment by RomLecat** [comment]
Hi,

I still get this issue (no red dot on Z version, recipe v3.3.5), and this is annoying because I can't switch to K version due to #859.
